cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -1909,6 +1909,110 @@ class SelectKUnstableMetaFunction : public MetaFunction {
   }
 };
 
+// ----------------------------------------------------------------------
+// Rank implementation
+
+const FunctionDoc rank_doc(
+    "Returns the ranking of an array",
+    ("This function computes a rank of the input array.\n"
+     "By default, Null values are considered greater than any other value and\n"
```

Comment:
Nit: no capitalization needed here :-)
```suggestion
     "By default, null values are considered greater than any other value and\n"
```